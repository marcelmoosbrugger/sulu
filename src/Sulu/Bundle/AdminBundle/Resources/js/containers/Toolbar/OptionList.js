// @flow
import classNames from 'classnames';
import React from 'react';
import Backdrop from '../../components/Backdrop';
import Option from './Option';
import optionListStyles from './optionList.scss';

type Props = {
    onOptionClick: (option: Object) => void,
    value?: string | number,
    size?: string,
    onClose?: () => void,
    options: Array<Object>,
};

export default class OptionList extends React.PureComponent<Props> {
    handleOptionClick = (option: Object) => {
        if (this.props.onOptionClick) {
            this.props.onOptionClick(option);
        }

        if (this.props.onClose) {
            this.props.onClose();
        }
    };

    handleBackdropClick = () => {
        if (this.props.onClose) {
            this.props.onClose();
        }
    };

    render() {
        const {
            size,
            value,
            options,
        } = this.props;
        const optionListClasses = classNames({
            [optionListStyles.optionList]: true,
            [optionListStyles[size]]: size,
        });

        return (
            <div>
                <ul className={optionListClasses}>
                    {
                        options.map((option, index: number) => {
                            const isSelected = option.value ? option.value === value : false;

                            return (
                                <Option
                                    key={index}
                                    size={size}
                                    value={option}
                                    label={option.label}
                                    disabled={option.disabled}
                                    selected={isSelected}
                                    onClick={this.handleOptionClick} />
                            );
                        })
                    }
                </ul>
                <Backdrop isOpen={true} onClick={this.handleBackdropClick} isVisible={false} />
            </div>
        );
    }
}
