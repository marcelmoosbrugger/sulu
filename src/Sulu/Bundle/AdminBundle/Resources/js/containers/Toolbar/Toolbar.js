// @flow
import Item from './Item';
import React from 'react';
import {observer} from 'mobx-react';
import toolbarStore from './stores/ToolbarStore';
import toolbarStyles from './toolbar.scss';

@observer
export default class Toolbar extends React.PureComponent {
    render() {
        return (
            <header className={toolbarStyles.toolbar}>
                <nav>
                    {toolbarStore.items.map((item) => (<Item key={item.title} {...item} />))}
                </nav>
            </header>
        );
    }
}
