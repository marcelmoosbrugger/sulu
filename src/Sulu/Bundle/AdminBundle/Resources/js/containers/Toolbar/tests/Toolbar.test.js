/* eslint-disable flowtype/require-valid-file-annotation */
import React from 'react';
import Toolbar from '../Toolbar';
import {render} from 'enzyme';
import toolbarStore from '../stores/ToolbarStore';

jest.mock('../stores/ToolbarStore', () => ({}));

test('Render the items from the ToolbarStore', () => {
    toolbarStore.items = [
        {
            title: 'Save',
            enabled: false,
            icon: 'save',
        },
        {
            title: 'Delete',
            icon: 'delete',
        },
    ];
    const view = render(<Toolbar />);
    expect(view).toMatchSnapshot();
});
