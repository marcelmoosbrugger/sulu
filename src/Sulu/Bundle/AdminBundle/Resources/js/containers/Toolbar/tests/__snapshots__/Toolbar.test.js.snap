// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the items from the ToolbarStore 1`] = `
<header
  class="toolbar"
>
  <nav>
    <button
      class="item"
      disabled=""
    >
      <span
        aria-hidden="true"
        class="icon fa fa-save"
      />
      <span
        class="title"
      >
        Save
      </span>
    </button>
    <button
      class="item"
    >
      <span
        aria-hidden="true"
        class="icon fa fa-delete"
      />
      <span
        class="title"
      >
        Delete
      </span>
    </button>
  </nav>
</header>
`;
